modules/internal/ArrayViewRankChange.chpl

```diff
@@ -280,15 +571,15 @@ module ArrayViewRankChange {
       // see what subset of target locales the rank-change slice hit vs.
       // not.
       //
-      compilerWarning("Calls to .targetLocales() on rank-change slices may currently return a superset of the locales targeted.");
+      //      compilerWarning("Calls to .targetLocales() on rank-change slices may currently return a superset of the locales targeted.");
       return arr.dsiTargetLocales();
     }
 
     proc dsiHasSingleLocalSubdomain() param
-      return privDom.dsiHasSingleLocalSubdomain();
+      return privDom.updom.dsiHasSingleLocalSubdomain();
```

Comment:
What is here may well be correct but I'm missing something. I'm wondering in particular why ArrayViewRankChangeDom.dsiHasSingleLocalSubdomain doesn't return the property on updom, since updom is the "user-facing" domain we are working with. Is it because updom isn't distributed, but perhaps the rest is? Then shouldn't this routine on the array return privDom.dsiHasSingleLocalSubdomain() i.e. consider what the original rank change is over instead of always returning `true`  (or whatever DefaultRectangularDom does with this method) ?